File: src/main/java/com/dawdawich/utils/PhotoQueue.java
package com.dawdawich.utils;

import org.telegram.telegrambots.api.methods.PartialBotApiMethod;

import java.util.ArrayDeque;

public class PhotoQueue {

    private ArrayDeque<PartialBotApiMethod> photos = new ArrayDeque<>();

    public PhotoQueue() {

    }


    public int addPhoto(PartialBotApiMethod photo) {
        photos.add(photo);
        return photos.size();
    }

    public PartialBotApiMethod getPhoto() {
        return photos.poll();
    }

    public int getSize() {
        return photos.size();
    }


}

File: src/main/java/com/dawdawich/utils/PhotoQueue.kt
package com.dawdawich.utils

import org.telegram.telegrambots.api.methods.PartialBotApiMethod

import java.util.ArrayDeque

class PhotoQueue {

    private val photos = ArrayDeque<PartialBotApiMethod<*>>()

    val photo: PartialBotApiMethod<*>?
        get() = photos.poll()

    val size: Int
        get() = photos.size

    fun addPhoto(photo: PartialBotApiMethod<*>): Int {
        photos.add(photo)
        return photos.size
    }
}
